hikari/impl/rest.py

```diff
@@ -441,15 +551,15 @@ def __init__(
         self._proxy_settings = proxy_settings
         self._lock = asyncio.Lock()
 
-        if token is None:
-            full_token = None
-        else:
-            if token_type is None:
-                token_type = _BOT_TOKEN_PREFIX
+        self._token: typing.Union[str, rest_api.TokenStrategy, None]
+        if isinstance(token, str):
+            self._token = f"{token_type.title()} {token}" if token_type else token
```

Comment:
I don't this we should allow passing the token type into `token`

Reply:
token type can't be None anymore

Reply:
You are still allowing passing `token_type` as `None` and checking if its `None` here :thinking:

Reply:
> I don't this we should allow passing the token type into token
BTW what do you mean by this

Reply:
You are allowing `token_type` being passed as `None` and, if its not there and the token is a string, taking the token as it is. This means that the `token_type` is already in the `token`, which I don't think is something we should allow

Reply:
well we don't get any extra functionality from enforcing that token type is passed separately and the old behaviour would've confused ppl who were passing the type as a part of the token when building a rest client since it'd also append a bot token to it making it invalid